eos-tech-support/eos-write-live-image

```diff
@@ -174,7 +175,7 @@ if [ $# -ne 1 ] ; then
     if [ $# -lt 1 ] ; then
         echo "Error: missing OUTPUT" >&2
     else
-        echo "Error: extra arguments after OUTPUT: $@" >&2
+        echo "Error: extra arguments after OUTPUT:" "$@" >&2
```

Comment:
What does this do?

Reply:
```
echo "Arguments: $@"
                 ^-- SC2145: Argument mixes string and array. Use * or separate argument.
```

In this case there's not much in it, given this is just error output – I picked one arbitrarily.

Reply:
https://github.com/koalaman/shellcheck/wiki/SC2145

I'd never thought about that. Here we're just passing to `echo`, so who cares how the arguments are split. But, I can see this being interesting in some other cases depending on exactly what you're trying to do with `$@` or some other array.
```
$ set -- "a b" c
$ a=("args: $@")
$ declare -p a
declare -a a='([0]="args: a b" [1]="c")'
$ a=("args: $*")
$ declare -p a
declare -a a='([0]="args: a b c")'
$ a=("args:" "$@")
$ declare -p a
declare -a a='([0]="args:" [1]="a b" [2]="c")'
```

Reply:
Interesting! I find bash's array handling deeply mysterious at the best of times, and definitely hadn't thought about this case.